ci_framework/roles/use_install_yamls/library/create_env_file.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2023, Chandan Kumar <chkumar@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: create_env_file
+
+short_description: It will create environment file containing all
+the variables exported for install_yamls makefile repo
+
+# If this is part of a collection, you need to use semantic versioning,
+# i.e. the version is of the form "2.5.0" and not "2.4".
+version_added: "1.0.0"
+
+description: It will create environment file containing all
+the variables exported for install_yamls makefile repo.
+
+options:
+    makefile_path:
+        description: Full path of the install_yamls makefile.
+        required: true
+        type: str
+    install_yamls_var:
+        description:
+            - List of exported install_yamls variable
+        required: true
+        type: dict
+    envfile_name:
+        description: name of the environment variable filename
+        required: false
+        type: string
+    check_var:
+        description: Check the environment var exists in the makefile
+        required: false
+        type: bool
+    create_yaml:
+        description: Create Ansible Environment Yaml file
+        required: false
+        type: bool
+author:
+    - Chandan Kumar (@raukadah)
+'''
+
+EXAMPLES = r'''
+# Create env var file
+- name: Create envfile for
+  become: true
+  create_env_file:
+    makefile_path: <path to make file>
+    install_yamls_var:
+        microshift: 1
+'''
+
+RETURN = r'''
+sucess:
+    description: env file created successfully
+    type: bool
+    returned: always
+    sample: "True"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+import yaml
+
+# Check install_yamls makefile exists
+def _check_makefile(makefile_path):
+    """
+    Check whether install_yamls makefile exists or not
+    """
+    if os.path.exists(makefile_path):
+        return True
+    else:
+        return False
+
+# Return install_yamls vars
+def _get_makefile_vars(makefile_path):
+    if _check_makefile(makefile_path):
```

Comment:
I would not create a function that is being used just once, but if it is for readability, it is fine.